tests/functional/lib/mesh/climesh.go

```diff
@@ -345,27 +345,32 @@ func NewCLIMeshFromYaml(MeshDefinition YamlData, dirPrefix string) (*CLIMesh, er
 					csvName, ok := vMap["service"]
 					if ok {
 						if csvName == "control" {
-							controlSocket = vMap["filename"].(string)
+							_, ok = vMap["filename"].(string)
+							if ok {
+								return nil, fmt.Errorf("control-service definition should not specify a filename")
+							}
+							controlServiceIndex = index
 							needsControlService = false
 						}
 					}
 				}
 			}
 		}
-		if !needsControlService {
-			node.controlSocket = controlSocket
-			continue
-		}
 		tempdir, err := ioutil.TempDir(ControlSocketBaseDir, "")
 		if err != nil {
 			return nil, err
 		}
-		node.controlSocket = filepath.Join(tempdir, "controlsock")
-		controlServiceYaml := make(map[interface{}]interface{})
+		controlSocket := filepath.Join(tempdir, "controlsock")
+		node.controlSocket = controlSocket
 		tmp := make(map[interface{}]interface{})
-		tmp["filename"] = node.controlSocket
-		controlServiceYaml["control-service"] = tmp
-		MeshDefinition.Nodes[k].Nodedef = append(MeshDefinition.Nodes[k].Nodedef, controlServiceYaml)
+		tmp["filename"] = controlSocket
```

Comment:
this `tmp` variable can go in the `if` now (as well as line 365 where we create it), makes it a bit more clear that that's only used when we are creating the whole controlservice definition.

Reply:
fixed!